src/v/raft/consensus.cc

```diff
@@ -243,7 +243,14 @@ ss::future<> consensus::stop() {
     co_await _event_manager.stop();
     co_await _append_requests_buffer.stop();
     co_await _batcher.stop();
-    co_await _bg.close();
+
+    auto f = _bg.close();
+    /**
+     * acquire lock after closing the gate, this way after
+     */
+    auto u = co_await _op_lock.get_units();
```

Comment:
If lock-holders also hold the gate, isn't it sufficient to just `broken()` the lock and then proceed to wait for the gate?  I would have thought we only need to take the lock here if there are other fibers that are taking the lock but not inside the gate.

Reply:
exactly, there are other fibers that acquire a lock outside a gate, calling `broken()` would be sufficient, but my intention was to finish gracefully where possible.

Reply:
i think i will simply refactor it to just break the semaphore, it will be simpler